pkgs/shells/zsh/default.nix

```diff
@@ -42,7 +42,7 @@ stdenv.mkDerivation {
     "--enable-multibyte"
     "--with-tcsetpgrp"
     "--enable-pcre"
-    "--enable-zprofile=${placeholder "out"}/etc/zprofile"
+    "--enable-zshenv=${placeholder "out"}/etc/zshenv"
```

Comment:
We would also need to move `/etc/zprofile` in the zsh module to `/etc/zshenv`.

Reply:
> We would also need to move /etc/zprofile in the zsh module to /etc/zshenv and deprecate its use there in case people have added to environment.etc.zprofile.text i.e. by doing a warning.

I do not understand what you suggest doing and the reason behind it.  Could you elaborate them?

> This file currently also sources /etc/zprofile.local which would need to deprecate with a warning as well.

Why does that file need to be deperated?

Reply:
@testplayername  I do not think you understand the init files **nixpkgs's zsh** reads correctly.

Without this patch, zsh reads:

1. `/etc/zshenv`
2. .zshenv of users'
3. `/nix/store/hash-zsh-version/etc/zprofile` (ignore that it will be compiled to zprofile.zwc).  In `/nix/store/hash-zsh-version/etc/zprofile`, zsh reads `/etc/zprofile` if that exists and otherwise `/etc/profile` on nixos; on non-nixos, zsh reads `/etc/zprofile`.
4. some more init files

With this patch, zsh reads:

1. `/nix/store/hash-zsh-version/etc/zshenv` (ignore that it will be compiled to zshenv.zwc).  In `/nix/store/hash-zsh-version/etc/zshenv`, zsh reads `/etc/zshenv` if that exists and otherwise `/etc/profile` on nixos; on non-nixos, zsh reads `/etc/zshenv`.
2. .zshenv of users'
3. `/etc/zprofile`
4. some more init files

Reply:
Also, please note that `$out/etc/zprofile` generates `/nix/store/hash-zsh-version/etc/zprofile` and `environment.etc.zprofile` generates `/etc/zprofile`.  They are different files.

Reply:
Ok. Still seems to work.